Wrapping/Generators/Python/itk/__init__.py

```diff
@@ -45,82 +46,55 @@ def _get_lazy_attributes(local_lazy_attributes, l_module, l_data):
                 # snake case always appended to end
                 local_lazy_attributes.setdefault(function, []).append(l_module)
 
-    import itkBase
-    import itkConfig
-    import itkLazy
+    from .support import itkBase as _itkBase
+    from .support import itkLazy as _itkLazy
+    from .conf import itkConfig as _itkConfig
     import sys
 
-    if itkConfig.LazyLoading:
+    if _itkConfig.LazyLoading:
         # If we are loading lazily (on-demand), make a dict mapping the available
         # classes/functions/etc. (read from the configuration modules) to the
         # modules they are declared in. Then pass that dict to a LazyITKModule
         # instance and (later) do some surgery on sys.modules so that the 'itk'
         # module becomes that new instance instead of what is executed from this
         # file.
         lazy_attributes = {}
-        for module, data in itkBase.itk_base_global_module_data.items():
+        for module, data in _itkBase.itk_base_global_module_data.items():
             _get_lazy_attributes(lazy_attributes, module, data)
 
-        if isinstance(sys.modules[__name__], itkLazy.LazyITKModule):
+        if isinstance(sys.modules[__name__], _itkLazy.LazyITKModule):
             # Handle reload case where we've already done this once.
             # If we made a new module every time, multiple reload()s would fail
             # because the identity of sys.modules['itk'] would always be changing.
             sys.modules[__name__].__init__(__name__, lazy_attributes)
             del lazy_attributes
         else:
             # Create a new LazyITKModule
-            lzy_module = itkLazy.LazyITKModule(__name__, lazy_attributes)
-            # Set the __path__ attribute, which is required for this_module
-            # to be used as a package
-            setattr(lzy_module, "__path__", __path__)
-            setattr(lzy_module, "__spec__", __spec__)  # pytype: disable=name-error
```

Comment:
We still need these

Reply:
Please also install `dask` in the test environment to enabled the related tests.

Reply:
Done for my local environments.  

I don't know what to change for the CI environments.

Reply:
@hjmjohnson `__path__` and `__spec__` are still being marked as removed in the global diff's.

Reply:
@thewtex   

Sorry, it took me a moment to remember.   The new code is needed to lift *all* the existing symbols into the lzy_module before overriding the current module with the replacement module.

```diff
-            # Set the __path__ attribute, which is required for this_module
-            # to be used as a package
-            setattr(lzy_module, "__path__", sys.modules[__name__].__path__)
-            setattr(
-                lzy_module, "__spec__", sys.modules[__name__].__spec__
-            )  # pytype: disable=name-error
+
+            # Pre-existing attributes need to be propagated too!
+            # except for the lazy overridden elements
+            exclusion_copy_list = ["__name__", "__loader__", "__builtins__"]
+            for k, v in sys.modules[__name__].__dict__.items():    #<--- SEE HERE   This line includes both __spec__ and __path__
+                if k not in exclusion_copy_list:
+                    setattr(lzy_module, k, v)  #<-- This line sets the the values
+
+             # Now override the default  sys.modules[__name__] (__name__  == 'itk' )
+             sys.modules[__name__] = lzy_module
```

Previously this behavior was done hap-hazardly, and sometimes redundantly when lazy loading was not used.  The following is no longer needed!
```diff
-    # Regardless of how it was loaded, fill up the itk module with the ITK types
-    # and extras.
-    import itkTypes
-
-    for k, v in itkTypes.__dict__.items():
-        if k != "itkCType" and not k.startswith("_"):
-            setattr(sys.modules[__name__], k, v)
-    del itkTypes
-
-    from .conf import itkInitHelpers as _itkInitHelpers
-
-    for k, v in _itkInitHelpers.__dict__.items():
-        if not k.startswith("_"):
-            setattr(sys.modules[__name__], k, v)
-
-    from .conf import itkExtras as _itkExtras
-
-    for k, v in _itkExtras.__dict__.items():
-        if not k.startswith("_"):
-            setattr(sys.modules[__name__], k, v)
-
-    # --
-    # Needed to propagate symbol to itk.image from itkTemplate.image
-    from .conf import itkTemplate as _itkTemplate
-
-    setattr(sys.modules[__name__], "image", _itkTemplate.image)
-    setattr(sys.modules[__name__], "output", _itkTemplate.output)
-    # --
```

FYI:  I used a debugger to verify that both 

```python

for k, v in sys.modules[__name__].__dict__.items():
  if k not in exclusion_copy_list:
    print(k)                            # <--------------------------------MATT Added for debugging locally 
    setattr(lzy_module, k, v)

############## console output
__doc__
__package__
__spec__                            # <------------ Verified included
__path__                             # <------------ Verified included
__file__
__cached__
conf
__version__
support
re
Optional
Union
Dict
```

Thanks for your careful review, I really appreciate it.

Hans

Reply:
@hjmjohnson great, thanks for the clarification!

Those two attributes were required for some very tricky Python import behavior. Crossing fingers that this refactoring will improve that behavior overall :crossed_fingers: